syntax = "proto2";

package vehicle;


message VehicleParam {
    optional double front_edge_to_center = 3 [default = 3.524];
    optional double back_edge_to_center = 4 [default = 1.2];
    optional double left_edge_to_center = 5 [default = 1.1];
    optional double right_edge_to_center = 6 [default = 1.1];
  
    optional double length = 7 [default = 4.724];
    optional double width = 8 [default = 1.9];
    optional double height = 9 [default = 1.5];
  
    optional double min_turn_radius = 10 [default = 4.9];
    optional double max_acceleration = 11 [default = 8];
    optional double max_deceleration = 12 [default = -8];
  
    // The following items are used to compute trajectory constraints in
    // planning/control/canbus,
    // vehicle max steer angle
    optional double max_steer_angle = 13 [default = 8.726646259971647];
    // vehicle max steer rate; how fast can the steering wheel turn.
    optional double max_steer_angle_rate = 14 [default = 8.55211];
    // vehicle min steer rate;
    optional double min_steer_angle_rate = 15 [default = nan];
    // ratio between the turn of steering wheel and the turn of wheels
    optional double steer_ratio = 16 [default = 17.14335423];
    // the distance between the front and back wheels
    optional double wheel_base = 17 [default = 2.72];
    // Tire effective rolling radius (vertical distance between the wheel center
    // and the ground).
    optional double wheel_rolling_radius = 18 [default = nan];
  
    // minimum differentiable vehicle speed, in m/s
    optional float max_abs_speed_when_stopped = 19 [default = 0.1];
  
    // minimum value get from chassis.brake, in percentage
    optional double brake_deadzone = 20 [default = nan];
    // minimum value get from chassis.throttle, in percentage
    optional double throttle_deadzone = 21 [default = nan];
}